### PR TITLE
Updated to Django Oscar 1.2.2

### PR DESCRIPTION
**Pull request by clintonb**
ECOM-7095

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce please review

**Comment by clintonb** [comment]
FYI @mjfrey note the removal of the custom messages.

File: ecommerce/extensions/basket/views.py [hunk 1]
```diff
@@ -317,62 +315,3 @@ def get_context_data(self, **kwargs):
...
-    View that applies a voucher to basket.
-    We change default messages oscar returns.
-    """
-
-    def form_valid(self, form):
-        super(VoucherAddMessagesView, self).form_valid(form)
-
-        code = form.cleaned_data['code']
-
-        for msg in list(messages.get_messages(self.request)):
```
Comment (mjfrey) [inline ecommerce/extensions/basket/views.py]: We need these messages. Why are they removed?

Comment (clintonb) [reply ecommerce/extensions/basket/views.py]: @mjfrey here are the default views now being used:
* https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/basket/views.py#L302
* https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/basket/views.py#L384

Eh...looks like they are still using the word voucher.

File: .travis.yml [hunk 2]
```diff
@@ -13,21 +13,28 @@ cache:
     - ecommerce/static/bower_components
 
 addons:
-    firefox: latest
```
Comment (MatthewPiatetsky) [inline .travis.yml]: why remove this?

Comment (clintonb) [reply .travis.yml]: See https://github.com/edx/ecommerce/pull/1146. Ignore the first commit.

File: Makefile [hunk 3]
```diff
@@ -32,7 +32,8 @@ help:
 
 requirements.js:
 	npm install
-	$(NODE_BIN)/bower install
+	# Allow root for Docker
```
Comment (MatthewPiatetsky) [inline Makefile]: what does this mean?

File: ecommerce/extensions/basket/views.py [hunk 4]
```diff
@@ -319,60 +318,75 @@ def get_context_data(self, **kwargs):
...
-                messages.error(
-                    self.request,
-                    _("Coupon code '{code}' is invalid.").format(code=code)
-                )
-
+                self.apply_voucher_to_basket(voucher)
         return redirect_to_referrer(self.request, 'basket:summary')
-
-
-class VoucherRemoveMessagesView(VoucherRemoveView):
```
Comment (mjfrey) [inline ecommerce/extensions/basket/views.py]: Unless this is fixed in the latest Oscar we still need this as we will start getting Server errors in the case that triggers this.

**Review by mjfrey — CHANGES_REQUESTED** [review]
Can we also see this in a sandbox or anywhere we can run some tests?

File: ecommerce/extensions/basket/views.py (hunk 4 above)
Comment (clintonb) [reply ecommerce/extensions/basket/views.py]: Under what conditions do these errors manifest? Django itself supports string lookups. I've never seen these issues.

Comment (mjfrey) [reply ecommerce/extensions/basket/views.py]: To reproduce:
1: Add item to basket, apply voucher.
2: Open second tab go to basket page.
3: First tab remove voucher.
4: Second tab remove voucher.
Server error will happen in Oscar.

Comment (clintonb) [reply ecommerce/extensions/basket/views.py]: Was an issue filed at https://github.com/django-oscar/django-oscar/?

Comment (mjfrey) [reply ecommerce/extensions/basket/views.py]: Not that I am aware of.

Comment (clintonb) [reply ecommerce/extensions/basket/views.py]: https://github.com/django-oscar/django-oscar/pull/2241

Comment (clintonb) [reply ecommerce/extensions/basket/views.py]: I'll restore this view for now; although, I don't think we should support multi-tab use cases.

Comment (mjfrey) [reply ecommerce/extensions/basket/views.py]: I agree, however we where getting Splunk alerts from this.

**Comment by clintonb** [comment]
@mjfrey sandbox building. Any other feedback?

**Review by mjfrey — APPROVED** [review]

**Pull request merged**